.github/workflows/preview.yml

```diff
@@ -0,0 +1,108 @@
+name: preview
+
+on:
+  workflow_call: ~
+
+permissions:
+  contents: read
+  packages: read
+  id-token: write
+  pull-requests: write
+  deployments: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Create Deployment
+        uses: actions/github-script@v7
+        id: deployment
+        with:
+          result-encoding: string
+          script: |
+            const response = await github.rest.repos.createDeployment({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                ref: "${{ github.event.pull_request.head.ref }}",
+                environment: `preview-${context.issue.number}`,
```

Comment:
Should this be preview-{repo}-{issue-number} so it's unique across different repos?